x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java

```diff
@@ -108,137 +104,161 @@ protected void masterOperation(
+                    dimensionFieldCaps.put(field, fieldCaps);
+                } else if (e.getValue().values().iterator().next().getMetricType() != null) {
+                    metricFieldCaps.put(field, fieldCaps);
+                }
+            }
+
             RollupActionRequestValidationException validationException = new RollupActionRequestValidationException();
-            if (fieldCapsResponse.get().size() == 0) {
-                validationException.addValidationError(
-                    "Could not find any fields in the index [" + originalIndexName + "] that were configured in job"
-                );
-                listener.onFailure(validationException);
-                return;
+            if (dimensionFieldCaps.isEmpty()) {
+                validationException.addValidationError("Index [" + sourceIndexName + "] does not contain any dimension fields");
             }
-            request.getRollupConfig().validateMappings(fieldCapsResponse.get(), validationException);
-            if (validationException.validationErrors().size() > 0) {
-                listener.onFailure(validationException);
-                return;
+            if (metricFieldCaps.isEmpty()) {
+                validationException.addValidationError("Index [" + sourceIndexName + "] does not contain any metric fields");
             }
 
-            // 2.
-            submitUnbatchedTask("rollup create index", new ClusterStateUpdateTask() {
+            final XContentBuilder mapping;
+            try {
+                mapping = createRollupIndexMapping(request.getRollupConfig(), dimensionFieldCaps, metricFieldCaps);
+            } catch (IOException e) {
+                listener.onFailure(e);
+                return;
+            }
+            CreateIndexClusterStateUpdateRequest createIndexClusterStateUpdateRequest = new CreateIndexClusterStateUpdateRequest(
+                "rollup",
+                rollupIndexName,
+                rollupIndexName
+            ).settings(
+                /*
+                 * When creating the rollup index, we copy the index.number_of_shards from source index,
+                 * and we set the index.number_of_replicas to 0, to avoid replicating the temp index.
+                 * We will set the correct number of replicas later.
+                 */
+                Settings.builder()
+                    .put(IndexMetadata.INDEX_HIDDEN_SETTING.getKey(), true)
+                    .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, sourceIndexMetadata.getNumberOfShards())
+                    .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                    .put(IndexMetadata.INDEX_ROLLUP_STATUS.getKey(), IndexMetadata.RollupTaskStatus.STARTED)
+                    .build()
+            ).mappings(XContentHelper.convertToJson(BytesReference.bytes(mapping), false, XContentType.JSON));
+
+            // 2. Create rollup index
+            submitUnbatchedTask("create-rollup-index", new ClusterStateUpdateTask() {
```

Comment:
I think maybe we can't use `submitUnbatchedTask`, right? The docs say " This method exists for legacy reasons but is deprecated and forbidden in new production code because unbatched tasks are a source of performance and stability bugs."

Reply:
I have seen the deprecation comment and researched a bit more into this. Method `clusterService.submitUnbatchedStateUpdateTask()` is deprecated by currently used all over the place. 

My goal was to get this PR merged and then work in this issue in a separate PR. 

@DaveCTurner do you have an opinion if we should merge this PR while using the deprecated method?

Reply:
I would very much prefer to avoid adding more unbatched tasks, they are truly awful for performance in large clusters. Here for instance if ILM were paused for a while then it might need to create thousands of rollup indices when restarted, and as currently written it would put all of those tasks in the queue at once and then work through them one-at-a-time.

That said, here it looks like we're just reworking an existing unbatched task rather than adding a new one right? If so this isn't really changing anything in terms of the cluster state tasks so it seems ok to proceed.

Reply:
> That said, here it looks like we're just reworking an existing unbatched task rather than adding a new one right? 

Actually now that I look closer it seems a bit more than that, we're introducing tasks that weren't there before which makes me concerned. Is there a good reason to avoid batching these these things? It shouldn't be too hard. See `MetadataIndexTemplateService#TEMPLATE_TASK_EXECUTOR` for a reasonably simple example.

Reply:
@DaveCTurner thanks for the feedback.

Reply:
Fixed in [332bef6](https://github.com/elastic/elasticsearch/pull/85708/commits/332bef648bbbddae58756acc40fca1adc9d3b257)